contracts/0.8.25/vaults/StakingVault.sol

```diff
@@ -185,7 +191,89 @@ contract StakingVault is IStakingVault, OwnableUpgradeable {
      * @notice Returns the address of `VaultHub`
      */
     function vaultHub() external view returns (address) {
-        return address(VAULT_HUB);
+        return _getStorage().vaultHub;
+    }
+
+    /**
+     * @notice Attaches the vault to a `VaultHub` and `Depositor`
+     * @param _vaultHub Address of `VaultHub`
+     * @param _depositor Address of `Depositor`
+     * @dev Can only be called by the owner
+     * @dev Reverts if `_vaultHub` is the zero address
+     * @dev Reverts if vault is already attached to VaultHub
+     */
+    function attachVaultHubAndDepositor(address _vaultHub, address _depositor) external onlyOwner {
```

Comment:
:warning: two-phase connect and disconnect are needed (`pendingConnect` and `pendingDisconnect`)

Reply:
idea: freshly connected vault has a valuation of its EL balance only (excluding CL balance) till the first report arrives